utils/weather_tools.py

```diff
@@ -37,114 +23,230 @@ def generate_stationID_dict():
+
+    Date = pd.date_range(str(year) + '-01-01', str(year) + '-12-31', tz = TZ_dict[StID_0])
+    df = pd.DataFrame(data=Date, columns=['Date'])     
+
     for frame in df_list:
         df = df.merge(frame, on='Date', how='left')
     df.set_index('Date', inplace=True)
-    df['MEAN_T'] = df.mean(axis=1)
-    df['M/D'] = df.index.strftime('%m/%d')
-    return df
+    df.to_csv("/Volumes/Nikhil hdd/Weather Data/" + f"{state}" +f"_{year}" + ".csv")
+    
+
+def all_states_files(year):
 
+    """Generate temeprature files for all the states for given year
+
+    returns files named ST_YYYY.csv """
+
+    states = pd.unique(df_stations_US.ST.values)
 
-def state_years(state, start_year, end_year):
-    """Mean temperature data for all years in range
+    for state in states:
+        
+        try:
+            path = "/Volumes/Nikhil hdd/Weather Data/" + f"{state}" + "_" + year + ".csv"
+            if not os.path.isfile(path):
+                print(state)
+                all_stations_for_state_year(state,year)
+        except ValueError:
+            print(f'Some issue with {state}')
 
+def states_timeseries():
 
-    Year range is (start_year, end_year) inclusive
+    """ concatenate files for a state from 2015-2019
 
-    Returns a single DataFrame
+    generates time series for each station in the state """
+
+    states = ['AL', 'AR', 'AZ', 'CA', 'CO', 'CT', 'FL', 'GA', 'IA', 'ID',
+ 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD', 'ME', 'MI', 'MN','MO', 'MS', 'MT', 'NC',
+ 'ND', 'NE', 'NH', 'NJ', 'NM', 'NV', 'NY', 'OH','OK', 'OR', 'PA', 'RI', 'SC',
+ 'SD', 'TN', 'TX', 'VA', 'VI', 'VT', 'WA', 'WI', 'WV', 'WY']
+
+    
+    years = range(2015,2020,1)
+    
+
+    
+
+    for state in states:
+        print(f'Processing {state}')
+
+        df_list = []
+
+        for year in years:
+
+            try:
+
+                filepath = f"/Volumes/Nikhil hdd/Weather Data/{year}_processed/{state}_{year}.csv"
+                df = pd.read_csv(filepath)
+                df_list.append(df)    
+
+            except FileNotFoundError as e:
+                print(e)
+
+        df = pd.concat(df_list, axis = 0)
+        # df = df_list[0]
+        df['MEAN_T'] = df.mean(axis = 1)*(9/5) + 32
+        
+        try:
+            df.rename(columns = {'Date': 'DATE'}, inplace = True)
+            df.set_index('DATE', inplace = True)
+        except KeyError as e:
+            print(e)
+            df.rename(index = {'Date': 'DATE'}, inplace = True)
+            
+        # df['MEAN_T'].to_csv(f"/Volumes/Nikhil hdd/Weather Data/state_data/{state}_2015-2019_raw.csv") 
+        df['MEAN_T'].to_csv(f"/Volumes/Nikhil hdd/Weather Data/state_data/{state}_2015-2019_raw.csv") 
+        
+
+
+
+
+
+file_st = f"{DATA_DIR}" + "/" + "states_info.csv"
+df_states = pd.read_csv(file_st, index_col = 0, parse_dates = ['LD'])
+regions = df_states.columns[2:-3]
+states = df_states.index.values
+states_in_region = {}
```

Comment:
As I mentioned before, global variables are evil. I understand these may _need_ to be global variables, but if so let's document it clearly, like
```
# global variables defined below
# variables hold data about states and whatnot
file_st = f"{DATA_DIR}" + "/" + "states_info.csv"
df_states = pd.read_csv(file_st, index_col = 0, parse_dates = ['LD'])
regions = df_states.columns[2:-3]
states = df_states.index.values
states_in_region = {}
```